repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeRouter.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.powerfilter
+
+import javax.inject.Named
+import javax.servlet.http.HttpServletResponse
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+
+// fixme: This class will serve as the basis of a rewrite of the routing component for Repose
+
+@Named
```

Comment:
One way of actually implementing this router would be to replace the router bean on configuration update and dropping the router factory completely.

I will also note that since this bean is in the `org.openrepose.powerfilter` package, it will exist per-node, and can access other node-level beans (e.g., node ID and cluster ID, which would be used to determine routing destinations).